osbuild/util/ostree.py

```diff
@@ -214,7 +216,41 @@ def parse_input_commits(commits):
     return commits["path"], data["refs"]
 
 
-def deployment_path(root: PathLike, osname: str, ref: str, serial: int):
+def parse_deployment_option(root, deployment):
+    """Parse the deployment option and return the osname, ref, and serial
+
+    The `deployment` arg contains the following sub fields:
+    - osname: Name of the stateroot used in the deployment (ie. fedora-coreos)
+    - ref: OStree ref to used for the deployment (ie. fedora/aarch64/coreos/next)
```

Comment:
Why is this here?  The ref is related to the origin, which is a distinct thing.  I guess it existed before this PR though?

Reply:
yeah. basically for backwards compat we have to keep around the old (`osname`, `ref`, `serial`) arguments.

I certainly don't plan to ever use those over `default` in the future. I honestly wish we could just get away with not requiring the deployment option to be specified at all in the future.

Reply:
Generally, the ref is used to get the commit ID to deploy https://github.com/osbuild/osbuild/blob/6b8c1872f64ce9eb173d51ddce9155bfbb98b5da/osbuild/util/ostree.py#L225-L226